server/src/main/java/org/opensearch/common/cache/tier/EhCacheDiskCachingTier.java

```diff
@@ -106,8 +127,17 @@ private EhCacheDiskCachingTier(Builder<K, V> builder) {
         this.DISK_WRITE_CONCURRENCY = Setting.intSetting(builder.settingPrefix + ".tiered.disk.ehcache.concurrency", 2, 1, 3);
         // Default value is 16 within Ehcache.
         this.DISK_SEGMENTS = Setting.intSetting(builder.settingPrefix + ".ehcache.disk.segments", 16, 1, 32);
-        this.cacheManager = buildCacheManager();
+
+        // In test cases, there might be leftover cache managers and caches hanging around, from nodes created in the test case setup
+        // Destroy them before recreating them
+        close();
```

Comment:
I don't think to solve for tests cases issues, we should make such changes in actual code. If there is an issue with test case, better to call close() after every test method?

Reply:
Just calling close() wouldn't work since the problem happens before the disk tier is fully instantiated (for the same reason as the above comment). But if we did the UUID thing, that might solve this problem as well